cmd/kube-proxy/app/server.go

```diff
@@ -126,15 +131,25 @@ func NewProxyConfig() *ProxyServerConfig {
 
 func NewProxyServer(
 	config *ProxyServerConfig,
+	client *kubeclient.Client,
+	endpointsConfig *proxyconfig.EndpointsConfig,
+	endpointsHandler proxyconfig.EndpointsConfigHandler,
 	iptInterface utiliptables.Interface,
+	oomAdjuster *oom.OOMAdjuster,
 	proxier proxy.ProxyProvider,
 	recorder record.EventRecorder,
+	serviceConfig *proxyconfig.ServiceConfig,
 ) (*ProxyServer, error) {
 	return &ProxyServer{
-		Config:       config,
-		IptInterface: iptInterface,
-		Proxier:      proxier,
-		Recorder:     recorder,
+		Client:           client,
+		Config:           config,
+		EndpointsConfig:  endpointsConfig,
```

Comment:
So if I understand, none of these new fields are used by kube-proxy, but ONLY by the Hollow version - is that right?  Some of them don't appear to be used at all - OOMAdjuster, EndpointsHandler, Client.  Why store them in this struct if they are never used?  The others - EndpointsConfig, ServiceConfig - are injected by Hollow, so no need to store them either.

I feel like I am missing something